hd-wallet/src/main/java/com/bloxbean/cardano/hdwallet/Wallet.java

```diff
@@ -0,0 +1,298 @@
+            DerivationPath derivationPath = DerivationPath.createExternalAddressDerivationPathForAccount(account);
+            derivationPath.getIndex().setValue(index);
+            return new Account(this.network, this.mnemonic, derivationPath);
+        } else {
+            if(cache.containsKey(index)) {
+                return cache.get(index);
+            } else {
+                Account acc = new Account(this.network, this.mnemonic, index);
+                cache.put(index, acc);
+                return acc;
+            }
+        }
+    }
+
+    /**
+     * Setting the current account for derivation path.
+     * Setting the account will reset the cache.
+     * @param account
+     */
+    public void setAccount(int account) {
+        this.account = account;
+        // invalidating cache since it is only held for one account
+        cache = new HashMap<>();
+    }
+
+    /**
+     * Returns the RootkeyPair
+     * @return
+     */
+    @JsonIgnore
+    public HdKeyPair getRootKeyPair() {
+        if(rootKeys == null) {
+            HdKeyGenerator hdKeyGenerator = new HdKeyGenerator();
+            try {
+                byte[] entropy = MnemonicCode.INSTANCE.toEntropy(this.mnemonic);
+                rootKeys = hdKeyGenerator.getRootKeyPairFromEntropy(entropy);
+            } catch (MnemonicException.MnemonicLengthException | MnemonicException.MnemonicWordException |
+                     MnemonicException.MnemonicChecksumException e) {
+                throw new WalletException("Unable to derive root key pair", e);
+            }
+        }
+        return rootKeys;
+    }
+
+    /**
+     * Finds needed signers within wallet and signs the transaction with each one
+     * @param txToSign
+     * @return signed Transaction
+     */
+    public Transaction sign(Transaction txToSign, Set<WalletUtxo> utxos) {
+        Map<String, Account> accountMap = utxos.stream()
+                .map(WalletUtxo::getDerivationPath)
+                .filter(Objects::nonNull)
+                .map(derivationPath -> getAccount(
+                        derivationPath.getAccount().getValue(),
+                        derivationPath.getIndex().getValue()))
+                .collect(Collectors.toMap(
+                        Account::baseAddress,
+                        Function.identity(),
+                        (existing, replacement) -> existing)); // Handle duplicates if necessary
+
+        var accounts = accountMap.values();
+
+        if(accounts.isEmpty())
+            throw new WalletException("No signers found!");
+
+        for (Account signerAcc : accounts)
+            txToSign = signerAcc.sign(txToSign);
+
+        return txToSign;
+    }
+
+//
+//    /**
+//     * Returns a list with signers needed for this transaction
+//     *
+//     * @param tx
+//     * @param utxoSupplier
+//     * @return
+//     */
+//    public List<Account> getSignersForTransaction(Transaction tx, WalletUtxoSupplier utxoSupplier) {
```

Comment:
This function isn't needed? Or will it added later again?

Reply:
@Kammerlo This is not needed currently, but I have kept it commented out for now in case it is required for any edge scenarios.